name.abuchen.portfolio/src/name/abuchen/portfolio/money/Values.java

```diff
@@ -320,33 +340,44 @@ public String format(Integer amount)
     };
 
     private final String pattern;
-    private final double divider;
     private final int factor;
+    private final double divider;
+    private final int precision;
     private final BigDecimal bdFactor;
 
-    private Values(String pattern, double divider, int factor)
+    private Values(String pattern, int factor)
     {
         this.pattern = pattern;
-        this.divider = divider;
         this.factor = factor;
+        this.divider = factor;
+        this.precision = (int) Math.log10(divider);
```

Comment:
Isn’t there (in principle) a risk that the `log10` might return something like 7.99999999, which is then rounded to 7 by the cast? Adding, e.g., 0.1 before casting might be safer.

Reply:
Thanks. I turned it around now. I configure the precision, and then calculate the factor with ```BigInteger.TEN.pow(precision).intValue()```